mlir/utils/jenkins/miopen-tests/miopen_validate.sh

```diff
@@ -136,6 +136,7 @@ function clean_miopen_caches() {
 function setup_environment() {
     export MIOPEN_FIND_MODE=1
     export MIOPEN_DRIVER_USE_GPU_REFERENCE=1
+    export MIOPEN_COMPILE_PARALLEL_LEVEL=1
```

Comment:
This needs to be reverted for final merge: MIOPEN_COMPILE_PARALLEL_LEVEL=1 is a debugging flag (that alters tuning behavior) and should not be used in production.

Reply:
This was added in 91e318b as a measure to get rid of errors encountered in the tuning process. Will keep an eye on it.

Reply:
Per discussion with @jerryyin , another temporarily measure is to modify the semantic of `miirLowerTuningParams`, which is used by `IsApplicable` in MIOpen MLIR solvers to determine if a tuning parameter is valid, so that `miirLowerTuningParams` would behave exactly as `miirLowerBin` for the whole lowering pipeline.

The rationale why `MIOPEN_COMPILE_PARALLEL_LEVEL=1` was introduced was because it was identified that:

1) without the environment variable, the tuning process would crash upon receiving a failure in getting a kernel binary blob from `miirLowerbin`. 
2) `miirLowerTuningParams` was invoked prior to `miirLowerBin` to help determine whether a tuning parameter is valid. It does NOT run through the whole lowering pipeline.
3) Should anything bad happens in the later part of the lowering pipeline, `miirLowerTuningParams` couldn't catch that.
4) MIOpen, as a client of `miirLowerTuningParams`, upon receiving the a success return value from the function, would assume `miirLowerBin` would definitely yield a working kernel, which is NOT always the case. Something bad might happen in the later part of the lowering pipeline. It could be in MIOpen dialect, could be in LLVM core, could be in LLVM AMDGPU backend.
5) The exception handling behavior inside MIOpen when `MIOPEN_COMPILE_PARALLEL_LEVEL=1` is NOT set is UNDEFINED. And currently it would just crash the tuning process.

Reply:
Commit #41b0a81 incorporate this temporary measure.

Reply:
I have just reverted the commit. Empirical testing shows the crash would still occur. And indeed it's inside MIOpen, albeit in a different place.

Reply:
I'll keep `MIOPEN_COMPILE_PARALLEL_LEVEL=1` in the test script for now.

After getting a deeper understanding of the other issue (tuning for fp16 backward data / weight) then revisit this topic.

Reply:
Thanks for confirming. As you want to proceed with merging and revisit those topics anyways I think you can just disable fp16 tuning in general as a temporary measure (Either in bash script or in Jenkinsfile). Otherwise we will be under trial-and-error mode for the code path untested (and probably shouldn't be there). Also, if you decide to disable fp16 tuning, please do file ticket to document all issues in independent tickets. This way we have a way to track those issues. 

The concern I had with all the temporary measures that got to merge with this PR is that it may another implicit deadline: the next ROCm release. On the other hand, if you keep the kpack in a separate branch then it will not impact anything in the mainline. Compared to the two, I actually think keeping it in the separate branch will give us more flexibility as it is unknown how much effort is needed to root cause all those issues.

Reply:
In the latest implementation, this hack is no longer necessary.